src/write.c

```diff
@@ -2641,15 +2759,22 @@ static avifResult avifEncoderWriteMiniBox(avifEncoder * encoder, avifRWStream *
     if (hasIcc) {
         AVIF_CHECKRES(avifRWStreamWriteBits(s, (uint32_t)image->icc.size - 1, fewMetadataBytesFlag ? 10 : 20)); // unsigned int(few_metadata_bytes_flag ? 10 : 20) icc_data_size_minus1;
     }
-    // if (hdr_flag && gainmap_flag && tmap_icc_flag)
-    //     unsigned int(few_metadata_bytes_flag ? 10 : 20) tmap_icc_data_size_minus1;
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+    if (hasHdr && hasGainmap && tmapIccSize != 0) {
+        const uint32_t tmapIccDataSize = (uint32_t)encoder->data->altImageMetadata->icc.size;
```

Comment:
If `tmapIccSize` is nonzero, I think it is equal to `encoder->data->altImageMetadata->icc.size`. So at this point, `tmapIccSize` and `tmapIccDataSize` have the same value. I think we can just use `tmapIccSize` directly.

Note: The comment at line 2818 below is similar to this one.